### DISCUSSION TITLE
Dvc_DiscreteOut

### DISCUSSION THREAD

**Discussion by ApolloPLC** [JeremyMedders/LogixLibraries#5] [General]
I was running some test on this AOI.
I like the function and versatility.

However, there seems to be an issue when using the pulse function.
When initializing using .PCmd_Pulse momentary command, .Out pulses for the defined periods of time. (Good)
However, if a .PCmd_Off  momentary command is given, .Out stays in its current state. 
So if .PCmd_Pulse=0, .Out=1 and .PCmd_Off=1, .Out will stop pulsing but remain =1.
When looking through the code .Pulse is =1.

After

if (PGOnOff.Out) then
	if (PulseOnOff) then
		Pulse := NOT Pulse;
	end_if;
end_if;

I added

if (Pulse & Not PulseOnOff) then
	Pulse := 0;
	end_if;
  
Seems to work fine, but I might be missing something or there might be a better way.
Thoughts?

**Comment by JeremyMedders** [comment]
Good catch!

I fixed the issue and just pushed v1.1 of the AOI.
